fullstack-cert/typescript-projects/flashcard-quiz-app/src/script.ts

```diff
@@ -0,0 +1,226 @@
+const cardDisplay = document.querySelector("#current-card") as HTMLElement;
+const cardButtonsContainer = document.getElementById(
+  "cards-list",
+) as HTMLElement;
+
+const frontInput = document.getElementById("front-text") as HTMLTextAreaElement;
+const backInput = document.getElementById("back-text") as HTMLTextAreaElement;
+
+let currentCardIndex = -1;
+let showingFront = true;
+let currentCards: FlashCard[] = [];
+
+interface FlashCard {
+  frontText: string;
+  backText: string;
+}
+
+class InvalidUserInputError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "InvalidUserInputError";
+    Object.setPrototypeOf(this, InvalidUserInputError.prototype);
+  }
+}
+
+function refresh(): void {
+  if (currentCardIndex < 0 || !currentCards[currentCardIndex]) {
+    cardDisplay.textContent =
+      "No card selected. Add a new card to get started!";
+    return;
+  }
+
+  const card = currentCards[currentCardIndex];
+  cardDisplay.querySelector(".card-front").textContent = card.frontText;
+  cardDisplay.querySelector(".card-back").textContent = card.backText;
+  // add correct background to current card
+  Array.from(cardButtonsContainer.children).forEach((child, i) => {
+    const button = child as HTMLElement;
+    if (i === currentCardIndex) {
+      button.classList.add("selected");
+    } else {
+      button.classList.remove("selected");
+    }
+  });
+}
+
+function flipCard(): void {
+  if (currentCardIndex === -1) return;
+  showingFront = !showingFront;
+  refresh();
+}
+
+function deleteCard(): void {
+  if (currentCardIndex === -1) {
+    console.log("No card selected to delete.");
+    return;
+  }
+
+  currentCards.splice(currentCardIndex, 1);
+  cardButtonsContainer.removeChild(
+    cardButtonsContainer.children[currentCardIndex],
+  );
+
+  Array.from(cardButtonsContainer.children).forEach((child, i) => {
+    (child as HTMLButtonElement).onclick = () => {
```

Comment:
Then we could use the typeguard here too.

Reply:
:heavy_check_mark: